Mathlib/SetTheory/Cardinal/Cofinality.lean

```diff
@@ -1114,6 +1117,35 @@ theorem sum_lt_of_isRegular {ι : Type u} {f : ι → Cardinal} {c : Cardinal} (
   sum_lt_lift_of_isRegular.{u, u} hc (by rwa [lift_id])
 #align cardinal.sum_lt_of_is_regular Cardinal.sum_lt_of_isRegular
 
+@[simp]
+theorem iUnion_subset_cardinal_lt {ι : Type u} {α : Type u} {t : ι → Set α} {c : Cardinal}
+    (h : #(⋃ i, t i) < c) (i : ι): #(t i) < c :=
+  lt_of_le_of_lt (Cardinal.mk_le_mk_of_subset <| subset_iUnion _ _) h
+
+@[simp]
+theorem card_iUnion_lt_iff_forall_of_isRegular {ι : Type u} {α : Type u} {t : ι → Set α}
+    {c : Cardinal} (hc : c.IsRegular) (hι : #ι < c) : #(⋃ i, t i) < c ↔ ∀ i, #(t i) < c := by
+  refine ⟨iUnion_subset_cardinal_lt, fun h ↦ ?_⟩
+  apply lt_of_le_of_lt (Cardinal.mk_sUnion_le _)
+  apply Cardinal.mul_lt_of_lt hc.aleph0_le
+    (lt_of_le_of_lt Cardinal.mk_range_le hι)
+  apply Cardinal.iSup_lt_of_isRegular hc (lt_of_le_of_lt Cardinal.mk_range_le hι)
+  simpa
+
+theorem biUnion_subset_cardinal_lt {α β : Type u} {s : Set α} {t : ∀ a ∈ s, Set β} {c : Cardinal}
+    (h : #(⋃ a ∈ s, t a ‹_›) < c) (a : α) (ha : a ∈ s) : # (t a ha) < c := by
+  rw [biUnion_eq_iUnion] at h
+  have := iUnion_subset_cardinal_lt h
+  simp_all only [iUnion_coe_set,
+    Subtype.forall]
+
+theorem card_biUnion_lt_iff_forall_of_isRegular {α β : Type u} {s : Set α} {t : ∀ a ∈ s, Set β}
+    {c : Cardinal} (hc : c.IsRegular) (hs : #s < c) :
+    #(⋃ a ∈ s, t a ‹_›) < c ↔ ∀ a (ha : a ∈ s), # (t a ha) < c := by
+  rw [biUnion_eq_iUnion, card_iUnion_lt_iff_forall_of_isRegular, SetCoe.forall']
+  · exact hc
+  · exact hs
```

Comment:
```suggestion
  rw [biUnion_eq_iUnion, card_iUnion_lt_iff_forall_of_isRegular hc hs, SetCoe.forall']
```
This is not just golfing: we don't like spawning subgoals as part of a rewrite (because it makes proofs harder to read).